src/main/java/com/example/pfairplayservice/common/util/MatchQueryBuilder.java

```diff
@@ -0,0 +1,98 @@
+package com.example.pfairplayservice.common.util;
+
+import com.example.pfairplayservice.model.enumfield.Status;
+
+import java.text.SimpleDateFormat;
+import java.util.Date;
+import java.util.List;
+
+public class MatchQueryBuilder {
```

Comment:
consider QueryDSL for this feature. http://www.querydsl.com
Implement two version. (string append vs query dsl)
compare them.